### PR TITLE
InteractiveTable: Add expand all to column

### PR DESCRIPTION
**Pull request by abannachGrafana**
## What is this feature?

- `<InteractiveTable />` component.
	- add optional expand all capability
- New icons for table expand and collapse all

## Why do we need this feature?

- Depending on your table needs if rows are expandable you may want to expand all of them.

## Who is this feature for?

- Users of the `<InteractiveTable />` component

## What issue(s) does this PR fix?

- https://github.com/grafana/grafana-adaptive-metrics-app/issues/247

Please check that:
- [ ] It works as expected from a user's perspective.
- [ ] If this is a pre-GA feature, it is behind a feature toggle.
- [ ] The docs are updated, and if this is a [notable improvement](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/#how-to-determine-if-content-belongs-in-a-whats-new-document), it's added to our [What's New](https://grafana.com/docs/writers-toolkit/writing-guide/contribute-release-notes/) doc.

expand-all-storybook

### CODE REVIEW

File: public/img/icons/unicons/table-collapse-all.svg
Comment (abannachGrafana) [inline public/img/icons/unicons/table-collapse-all.svg:1]: SVG provided by UX

File: public/img/icons/unicons/table-expand-all.svg
Comment (abannachGrafana) [inline public/img/icons/unicons/table-expand-all.svg:1]: SVG provided by UX

File: packages/grafana-ui/src/components/InteractiveTable/InteractiveTable.tsx [hunk 1]
```diff
@@ -146,6 +146,10 @@ interface Props<TableData extends object> {
    * Render function for the expanded row. if not provided, the tables rows will not be expandable.
    */
   renderExpandedRow?: (row: TableData) => ReactNode;
+  /**
+   * Whether to show the "Expand all" button. Depends on renderExpandedRow to be provided. Defaults to false.
+   */
+  showExpandAll?: boolean;
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/InteractiveTable/InteractiveTable.tsx:152]: Seeing as this prop depends on another one, we can make the type support for that a bit better and provide a type error when `showExpandAll` is used without `renderExpandedRow`:

```tsx
interface WithExpandableRow<TableData extends object> extends BaseProps<TableData> {
  renderExpandedRow: (row: TableData) => ReactNode;
  showExpandAll?: boolean;
}

interface WithoutExpandableRow<TableData extends object> extends BaseProps<TableData> {
  renderExpandedRow?: never;
  showExpandAll?: never;
}

type Props<TableData extends object> = WithExpandableRow<TableData> | WithoutExpandableRow<TableData>;
```

**Review by Clarity-89 — COMMENTED** [review]
Great work on this! 🎉  Left a few minor comments.

File: packages/grafana-ui/src/components/InteractiveTable/Expander/index.tsx [hunk 2]
```diff
@@ -27,3 +27,19 @@ export function ExpanderCell<K extends object>({ row, __rowID }: CellProps<K, vo
...
 }
+
+export function ExpanderHeader<K extends object>({ isAllRowsExpanded, toggleAllRowsExpanded }: HeaderProps<K>) {
+  return (
+    <div className={expanderContainerStyles}>
+      <IconButton
+        data-testid={'expand-all-button'}
+        aria-label={!isAllRowsExpanded ? 'Expand all rows' : 'Collapse all rows'}
+        name={!isAllRowsExpanded ? 'table-expand-all' : 'table-collapse-all'}
+        onClick={() => toggleAllRowsExpanded()}
```
[the remark this answers is not in the document]
Comment (abannachGrafana) [reply packages/grafana-ui/src/components/InteractiveTable/Expander/index.tsx:38]: Unfortunately there is a type mismatch if done this way. Onclick is passing a mouseevent and toggleAll expects a boolean.

File: packages/grafana-ui/src/components/InteractiveTable/InteractiveTable.tsx (hunk 1 above)
Comment (abannachGrafana) [reply packages/grafana-ui/src/components/InteractiveTable/InteractiveTable.tsx:152]: Nice!
image

File: packages/grafana-ui/src/components/InteractiveTable/InteractiveTable.tsx [hunk 3]
```diff
@@ -146,6 +147,10 @@ interface Props<TableData extends object> {
...
```
Comment (Clarity-89) [inline packages/grafana-ui/src/components/InteractiveTable/InteractiveTable.tsx:153]: You don't need these two props on `BaseProps` as they're included in their own interfaces.

**Review by Clarity-89 — APPROVED** [review]
One minor comment, otherwise looks good 🚀

**Pull request merged**